pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -783,3 +783,98 @@ func (s *Session) UpdateVirtualMachine(
 	// TODO: Find a better place for this?
 	return s.attachClusterModule(vmCtx, resVM, updateArgs.ResourcePolicy)
 }
+
+func vmUpdateGetArgs(
+	vmCtx context.VirtualMachineContext, client client.Client, updateArgs *VMUpdateArgs) (*VMUpdateArgs, error) {
+	vmClass, err := utils.GetVirtualMachineClass(vmCtx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	resourcePolicy, err := utils.GetVMSetResourcePolicy(vmCtx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	vmMD, err := utils.GetVMMetadata(vmCtx, client)
+	if err != nil {
+		return nil, err
+	}
+
+	updateArgs.VMClass = vmClass
+	updateArgs.VMMetadata = vmMD
+	if resourcePolicy != nil {
+		updateArgs.ResourcePolicy = resourcePolicy
+	} else {
+		// If resourcePolicy is not set, use default MinCPUFreq value.
+		updateArgs.MinCPUFreq = 0
```

Comment:
It is not trivial to infer how resource policy and CPU min frequency are related?  Can you add a comment here?  Ignore if you are just moving code around.

Reply:
Refactor this, not apply to the current implementation any more.